d/aws_kms_alias: Add target_key_arn attribute

bflad opened the pull request:
This is an alternate implementation to #2224. Closes #2009. 

KMS keys and aliases are bound to the same AWS partition, account, and region so it is safe to assemble the target key ARN from the alias ARN and target key ID. Reference: http://docs.aws.amazon.com/sdk-for-go/api/service/kms/#KMS.CreateAlias

> The alias and the CMK it is mapped to must be in the same AWS account and the same region. You cannot perform this operation on an alias in a different AWS account.

```
make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsKmsAlias'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccDataSourceAwsKmsAlias -timeout 120m
=== RUN   TestAccDataSourceAwsKmsAlias
--- PASS: TestAccDataSourceAwsKmsAlias (41.76s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	41.791s
```

radeksimko commented on aws/data_source_aws_kms_alias_test.go:
```diff
@@ -47,6 +48,19 @@ func testAccDataSourceAwsKmsAliasCheck(name string) resource.TestCheckFunc {
+				attr["target_key_arn"],
+				targetKeyArn,
+			)
+		}
```
How do you feel about `regexp` matching with `$` here? It seems that would be a little bit cleaner approach compared to modification of the original attribute.

starlton-eb replied:
This would be awesome to have right now. I racked by brain all day wondering why my aws_efs_file_system instances were being recreated and discovered that it was due to a recent update to our stack that includes looking up a kms key:

bflad commented:
I'll update and merge this PR after we release v1.7.1 (hopefully tomorrow). 🚀

bflad commented:
Rebased and updated if you'd like to give another look
```
make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsKmsAlias'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccDataSourceAwsKmsAlias -timeout 120m
=== RUN   TestAccDataSourceAwsKmsAlias
--- PASS: TestAccDataSourceAwsKmsAlias (40.72s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	40.769s
```

ColinHebert commented:
Helps/fixes #3019

bflad commented:
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
